app/models/user.rb

```diff
@@ -45,12 +46,27 @@ def send_password_reset
     UserMailer.forgot_password(id).deliver
   end
 
+  def recompute_credits
+    self.credits_count = self.credits.inject(0) { |sum, credit| sum += credit.amount }
+    save!
+  end
+
   private
 
   def password_set?
     return !(self.password.blank? && !self.new_record?)
   end
 
+  def add_verification_credits
+    credits.build({ amount: 5, creditable: User.find(0), description: 0 })
+    save
```

Comment:
1. Amount, ADMIN_USER should be a constant
2. What is 0 as description?
3. Why save is called as this is already in a callback?